File: src/server/ai/providers/openrouter.ts
import { AIAssistantConfig } from "../config";
import { OpenAICompatibleProvider } from "./base";

export const openRouterProvider = new OpenAICompatibleProvider(
  "openrouter",
  {
    apiKey: AIAssistantConfig.openRouterApiKey,
    baseUrl: "https://openrouter.ai/api/v1/chat/completions",
    model: AIAssistantConfig.openRouterModel,
    timeoutMs: AIAssistantConfig.requestTimeoutMs,
    maxRetries: AIAssistantConfig.maxRetries,
  }
);

File: ai-assistant/providers/openrouter.py
import os
from .base import OpenAICompatibleProvider

openrouter_provider = OpenAICompatibleProvider(
    name="openrouter",
    api_key=os.getenv("OPENROUTER_API_KEY", ""),
    base_url="https://openrouter.ai/api/v1/chat/completions",
    model=os.getenv("OPENROUTER_MODEL", "google/gemini-2.0-flash"),
)
